### DISCUSSION TITLE
Migrate from electron webview tag to electron BrowserView

### DISCUSSION THREAD

**Discussion by balajiv113** [ferdium/ferdium-app#819] [Ideas]
**Problem Description**
Currently the core code for displaying services is done by using webview tag.
https://www.electronjs.org/docs/latest/api/webview-tag#warning

Also the chromium source of webview tag state its deprecated
https://developer.chrome.com/docs/extensions/reference/webviewTag/

electron/electron#34356

**Affected Code**
All the usages of dependency react-electron-web-view is affected

**Proposed Solution**
The preferred solution would be to use BrowserView as a replacement for this
https://www.electronjs.org/docs/latest/api/browser-view

**Alternatives Considered**
Some other alternates would be,

using iframe (which doesn't look simpler)

**Comment by vraravam** [comment]
@balajiv113 - could you try to work on this before the internal server? This seems more pressing and will limit our ability to upgrade electron (core) itself.

**Reply by balajiv113** [reply]
@vraravam Sure we can plan something but this is going to be like a complete rewrite of ferdium core which will take more time.

About electron upgrade i believe after new year (jan 1st week) we should get some answers from electron team about the issue raised

**Reply by DAcodedBEAT** [reply]
Hello 👋 Is there an update on this?
